ankihub/main/suggestions.py

```diff
@@ -383,11 +383,13 @@ def _added_and_removed_tags(
 def _fields_that_changed(
     prev_fields: List[Field], cur_fields: List[Field]
 ) -> List[Field]:
-    result = [
-        cur_field
-        for cur_field, prev_field in zip(cur_fields, prev_fields)
-        if cur_field.value != prev_field.value
-    ]
+    result = []
+    for prev_field in prev_fields:
+        cur_field = next(
+            (field for field in cur_fields if field.name == prev_field.name), None
+        )
+        if cur_field and cur_field.value != prev_field.value:
+            result.append(cur_field)
```

Comment:
:pick: I think it would be better to undo this change. It's not needed now that both field lists contain all fields, including empty ones.

Reply:
We allow extra local fields though? What if the deck owner has extra fields they didn't publish yet?

Reply:
`to_note_data` only returns fields which are defined in the note type. 

The fields returned by `to_note_data` come from here:
https://github.com/AnkiHubSoftware/ankihub_addon/blob/6b1bfe24b9bebadafafdd66d87967bdec502cdec/ankihub/main/exporting.py#L48-L58